### PR TITLE
First draft of adaptation for Proemion

### PR DESCRIPTION
**Pull request by Scifire**
This is the first draft to adjust the module for our needs. 

@SISheogorath what do you think? 

child of: https://github.com/Proemion/puppet/issues/534

### CODE REVIEW

File: templates/ralus.cfg.erb [hunk 1]
```diff
@@ -9,8 +9,8 @@ Software\Symantec\Backup Exec For Windows\Backup Exec\Engine\Agents\Advertise No
 Software\Symantec\Backup Exec For Windows\Backup Exec\Engine\Agents\Advertisement Purge=0
 Software\Symantec\Backup Exec For Windows\Backup Exec\Engine\Agents\Advertising Disabled=0
 Software\Symantec\Backup Exec For Windows\Backup Exec\Engine\Agents\Advertising Interval Minutes=240
-Software\Symantec\Backup Exec For Windows\Backup Exec\Engine\Agents\Agent Directory List_1=<%= scope.lookupvar("::hostname") %>.bak.epost-dev.de
-Software\Symantec\Backup Exec For Windows\Backup Exec\Engine\Agents\Agent Directory List_2=EPD-BE2012
+Software\Symantec\Backup Exec For Windows\Backup Exec\Engine\Agents\Agent Directory List_1=fulitvm016.ad1.proemion.com #make this variable
+Software\Symantec\Backup Exec For Windows\Backup Exec\Engine\Agents\Agent Directory List_2=FULITVM016 #make this variable
```
Comment (SISheogorath) [inline templates/ralus.cfg.erb]: We might want to use variables here instead hardcoded hostnames?

File: manifests/init.pp [hunk 2]
```diff
@@ -2,9 +2,8 @@
 #
 # This class installs and configures a Symantec BackupExec agent on Linux
 #
-class backupexec (
-    $pkgname = $backupexec::params::pkgname,
-) inherits backupexec::params {
+class backupexec {
```
Comment (SISheogorath) [inline manifests/init.pp]: You shouldn't remove the `inherits` part, since this can provide nice defaults.

File: manifests/init.pp [hunk 3]
```diff
@@ -15,9 +14,9 @@
     require => Group['beoper']
   }
 
-  package { $pkgname:
-    ensure  => present,
-    require => User['beuser'],
+  package { 'vrtsralus':
```
Comment (SISheogorath) [inline manifests/init.pp]: Changing from the variable to the hardcoded name renders the params.pp useless. Not sure we want that.

File: manifests/init.pp [hunk 4]
```diff
@@ -26,7 +25,7 @@
     group   => 'beoper',
     mode    => '0644',
     content => template('backupexec/ralus.cfg.erb'),
-    require => Package[$backupexec::params::pkgname],
+    require => Package['vrtsralus'],
```
Comment (SISheogorath) [inline manifests/init.pp]: Same as with the package installation.

File: manifests/init.pp [hunk 5]
```diff
@@ -15,9 +14,9 @@
     require => Group['beoper']
   }
 
-  package { $pkgname:
-    ensure  => present,
-    require => User['beuser'],
+  package { 'vrtsralus':
+    ensure  => latest,
+    require => [ User['beuser'], Apt::Source['rm-apt-repo'] ],
```
Comment (SISheogorath) [inline manifests/init.pp]: Again, I don't think hardcoding our custom setup content in here is helpful.

**Review by SISheogorath — CHANGES_REQUESTED** [review]
We should keep the module generic. I would like to avoid hardcoding too many things towards our setup if we really want to appear useful to anyone else.

File: templates/ralus.cfg.erb (hunk 1 above)
Comment (Scifire) [reply templates/ralus.cfg.erb]: @SISheogorath yes that's the final plan, for the first draft i'd like to try with this

**Pull request merged**